samples/certs/Makefile

```diff
@@ -121,3 +133,52 @@ root-key.pem:
 	@echo "O = $(CITADEL_ORG)" >> $@
 	@echo "CN = $(CITADEL_CN)" >> $@
 	@echo "L = $(L:/=)" >> $@
+
+#------------------------------------------------------------------------
+##WORKLOAD_CERTS=1 <name>-certs: generate workload certificates using above intermediate ca Includes all PEM needed.
+
+%/workload-cert-chain.pem: %/workload-cert.pem %/ca-cert.pem root-cert.pem
+	@echo "generating $@"
+	@cat $^ > $@
+
+%/workload-cert.pem: %/workload.csr
+	@echo "generating $@"
+	@openssl x509 -req -days $(WORKLOAD_DAYS) \
+		-CA $(dir $<)/ca-cert.pem  -CAkey $(dir $<)/ca-key.pem -set_serial $(CITADEL_SERIAL) \
+		-extensions req_ext -extfile $(dir $<)/workload.conf \
+		-in $< -out $@
+
+%/workload.csr: L=$(dir $@)
+%/workload.csr: %/key.pem %/workload.conf
+	@echo "generating $@"
+	@openssl req -new -config $(L)/workload.conf -key $< -out $@
+
+%/key.pem:
+	@echo "generating $@"
+	@mkdir -p $(dir $@)
```

Comment:
I think `dir` can be removed throughout this makefile with the correct automatic variable.